packages/webapp/src/elections/components/ongoing-election-components/request-election-meeting-link-button.tsx

```diff
@@ -181,3 +147,77 @@ export const RequestElectionMeetingLinkButton = ({
         </Button>
     );
 };
+
+interface JoinMeetingButtonProps {
+    stage: RoundStage;
+    encryptedData: EncryptedData;
+}
+
+const JoinMeetingButton = ({
+    stage,
+    encryptedData,
+}: JoinMeetingButtonProps) => {
+    console.info(stage);
+
+    const [roundMeetingLink, setRoundMeetingLink] = useState("");
+    const [
+        failedToDecryptMeetingLink,
+        setFailedToDecryptMeetingLink,
+    ] = useState(false);
+
+    useEffect(() => {
+        decryptMeetingLink();
+    }, [encryptedData]);
+
+    const decryptMeetingLink = async () => {
+        if (encryptedData) {
+            try {
+                const encryptionKey = encryptedData.keys.find((k) =>
+                    getEncryptionKey(k.recipient_key)
+                );
+
+                if (!encryptionKey) {
+                    throw new Error(
+                        "Encryption key not found to decrypt the current meeting"
+                    );
+                }
+
+                const decryptedLink = await decryptPublishedMessage(
+                    encryptedData.data,
+                    encryptionKey.recipient_key,
+                    encryptionKey.sender_key,
+                    encryptionKey.key
+                );
+
+                setRoundMeetingLink(decryptedLink);
+            } catch (e) {
+                console.error("fail to decrypt meeting link", e);
+                onError(e.message);
+                setFailedToDecryptMeetingLink(true);
+            }
+        } else {
+            setRoundMeetingLink("");
+        }
+    };
+
+    return failedToDecryptMeetingLink ? (
+        <div className="flex items-center space-x-2">
+            <Text type="danger">
+                <BsExclamationTriangle className="mr-1 mb-px" />
+            </Text>
+            <Text type="danger">
+                Failed to retrieve the meeting link. Please reach out to someone
+                from your election round group or access the support.
```

Comment:
I would tweak this text to: "Failed to retrieve meeting link. Ask someone else in your election round group for the link or join the community support room above."